spec/tests/check.fmf

```diff
@@ -0,0 +1,36 @@
+summary: Additional test checks
+
+story:
+    As a tester I want to employ additional checks before, during
+    and after test execution. These checks would complement the
+    actual test by monitoring various aspects and side-effects of
+    the test execution.
+
+description: |
+    In some cases we want to run additional checks while running a
+    test. A nice example is a check for unexpected SELinux AVCs
+    produced during the test, this can point to additional issues
+    the user can run into. Another useful checks would be kernel
+    panic detection, core dump collection or collection of system
+    logs.
+
+    To control checks for a given test, use L1 ``check`` key.
```

Comment:
Explicitly describe the expected format as well? It must be a string, a list of strings or list of dicts...

Reply:
Aren't examples descriptive enough? I didn't bother, these things seem to be described by example in tmt docs.

Reply:
I would actually try to improve the documentation in general. Some keys do mention types, e.g. https://tmt.readthedocs.io/en/stable/spec/tests.html#component - "Must be a string or a list of strings." -, some don't at all, e.g. https://tmt.readthedocs.io/en/stable/spec/plans.html#environment. How about following the approach taken in https://tmt.readthedocs.io/en/stable/spec/plans.html#results-format and https://tmt.readthedocs.io/en/stable/spec/plans.html#guest-topology-format, i.e. text describing the field, without diving into details like types, then commented (code block) describing the syntax variants, followed by a set of real-life examples? I don't mind submitting patches to improve this area.

Maybe we could add a key into the `story` class to hold this "schema" type of info. It seems to be a common need, and it would give rendering engines (rst template or `tmt story show`...) a chance to display the syntax nicely formatted.

Reply:
> I would actually try to improve the documentation in general. Some keys do mention types, some don't at all.

I've quickly checked the spec and it seems that almost all test/plan/story keys have this documented, plan environment is an exception and should be fixed.

> How about ... describing the field, without diving into details like types, then commented (code block) describing the syntax variants, followed by a set of real-life examples?

Sound good, and works fine especially for more complex structures. On the other hand, for short statements like `Must be a string or a list of strings.` an extra code block sounds a bit like an overkill to me.

> Maybe we could add a key into the `story` class to hold this "schema" type of info.

Interesting idea, a dedicated field sounds tempting. Could be useful. And could be rendered in docs in a consistent way. Perhaps we would even find a key name general enough which could become a real story attribute. Perhaps, `requirement` or `definition` or `criteria` or something like that?

Let's cover that in a separate PR to not block this one?

Reply:
New PRs, please.